client/providers/CallProvider/CallProvider.tsx

```diff
@@ -143,23 +129,58 @@ export const CallProvider: FC = ({ children }) => {
 	);
 
 	useEffect(() => {
-		Notifications.onUser('callerjoined', handleQueueJoined);
-		Notifications.onUser('agentcalled', handleAgentCalled);
-		Notifications.onUser('agentconnected', handleAgentConnected);
-		Notifications.onUser('queuememberadded', handleMemberAdded);
-		Notifications.onUser('queuememberremoved', handleMemberRemoved);
-		Notifications.onUser('callabandoned', handleCallAbandon);
-		Notifications.onUser('call.callerhangup', handleCallHangup);
-		Notifications.onUser('call.callerhangup', handleCallHangup);
-	}, [
-		handleQueueJoined,
-		handleMemberAdded,
-		handleMemberRemoved,
-		handleCallAbandon,
-		handleAgentConnected,
-		handleCallHangup,
-		handleAgentCalled,
-	]);
+		if (!voipEnabled || !user) {
+			return;
+		}
+
+		const unsubscribeFromCallerJoined = subscribeToNotifyUser(`${user._id}/callerjoined`, handleQueueJoined);
+		return unsubscribeFromCallerJoined;
```

Comment:
![image](https://user-images.githubusercontent.com/2263066/156823017-40e02346-5a62-4321-b2a9-4acb80f6f190.png)

```suggestion
		return subscribeToNotifyUser(`${user._id}/callerjoined`, handleQueueJoined);
```